WooCommerce/WooCommerceUITests/UITests.xctestplan

```diff
@@ -43,6 +43,11 @@
         "value" : "$(BUILDKITE_MESSAGE)"
       }
     ],
+    "targetForVariableExpansion" : {
+      "containerPath" : "container:WooCommerce.xcodeproj",
+      "identifier" : "B56DB3C52049BFAA00D4AA8E",
+      "name" : "WooCommerce"
+    },
```

Comment:
CI doesn't complain about not having this, but Xcode does.

Reply:
I'm not sure how this `targetForVariableExpansion` works, but I'm wondering if this might be the reason why UI Tests analytics don't work, especially given that Unit Tests run on their own Xcode target (where they don't depend on the app target and should ideally be totally independent and isolated from the app target?) and use a dedicated test runner, but UI Tests use a Host target (the app) and run inside it like a plugin driving the app via accessibility APIs… so should that `targetForVariableExpansion` here be anything different for UI Tests vs Unit Tests given all that?

_(This is just a total guess and shot in the dark, as I've never played with XCTestPlans and never seen this `targetForVariableExpansion` property in my life before 😅)_

Reply:
My understanding of the `targetForVariableExpansion` is that it is a similar feature as what we can do with the scheme, where you can let "Test" use the same environment variables as defined in "Run", which helps with DRY.

I haven't had a chance to investigate more on this. For reference, the error I'm getting locally with the UI tests (using an additional env var `BUILDKITE_ANALYTICS_DEBUG_ENABLED`) is

```
[BuildkiteTestCollector] debug: uploading TestResults(format: "json", runEnv: Core.RunEnvironment(ci: Optional("buildkite"), key: "", url: Optional(""), branch: Optional(""), commitSha: Optional(""), number: Optional(""), jobId: Optional(""), message: Optional(""), debug: Optional("true"), version: Optional("0.1.1"), collector: Optional("test-collector-swift")), data: [Core.Trace(id: "68B6B3B6-FA71-4A70-9EE6-D3A70F3924CF", scope: Optional("ReviewsTests"), name: Optional("test_load_reviews_screen"), identifier: Optional("ReviewsTests.test_load_reviews_screen"), location: nil, fileName: nil, result: Optional(Core.Trace.Result.failed), failureReason: Optional("XCTAssertTrue failed - Page UITestsFoundation.PasswordScreen is not loaded."), failureExpanded: [Core.Trace.FailureExpanded(expanded: ["Assertion Failure at BaseScreen.swift:24: XCTAssertTrue failed - Page UITestsFoundation.PasswordScreen is not loaded."], backtrace: ["0 Address 4377259627 is in untracked image \'XCTestCore\'", "1 Address 4377322471 is in untracked image \'XCTestCore\'", "2 Address 4377322819 is in untracked image \'XCTestCore\'", "3 Address 4379884299 is in untracked image \'libXCTestSwiftSupport.dylib\'", "4 Address 4379883619 is in untracked image \'libXCTestSwiftSupport.dylib\'", "5 Address 4414629075 is in untracked image \'UITestsFoundation\'", "6 Address 4414629435 is in untracked image \'UITestsFoundation\'", "7 Address 4414629527 is in untracked image \'UITestsFoundation\'", "8 Address 4379869427 is in untracked image \'libXCTestSwiftSupport.dylib\'", "9 Address 4379869503 is in untracked image \'libXCTestSwiftSupport.dylib\'", "10 Address 7357804347 is in untracked image \'libswiftObjectiveC.dylib\'", "11 Address 4379869139 is in untracked image \'libXCTestSwiftSupport.dylib\'", "12 Address 4414628407 is in untracked image \'UITestsFoundation\'", "13 Address 4414627911 is in untracked image \'UITestsFoundation\'", "14 Address 4414508347 is in untracked image \'UITestsFoundation\'", "15 Address 4414507715 is in untracked image \'UITestsFoundation\'", "16 Address 4414467199 is in untracked image \'UITestsFoundation\'", "17 static LoginFlow.logInWithWPcom()", "18 ReviewsTests.setUpWithError()", "19 @objc ReviewsTests.setUpWithError()", "20 Address 4377472791 is in untracked image \'XCTestCore\'", "21 Address 4377540595 is in untracked image \'XCTestCore\'", "22 Address 4377540311 is in untracked image \'XCTestCore\'", "23 Address 4377274967 is in untracked image \'XCTestCore\'", "24 Address 4377539939 is in untracked image \'XCTestCore\'", "25 Address 4377540467 is in untracked image \'XCTestCore\'", "26 Address 4377472743 is in untracked image \'XCTestCore\'", "27 Address 4377472259 is in untracked image \'XCTestCore\'", "28 Address 4377472015 is in untracked image \'XCTestCore\'", "29 Address 4377242947 is in untracked image \'XCTestCore\'", "30 Address 4377471819 is in untracked image \'XCTestCore\'", "31 Address 4377359307 is in untracked image \'XCTestCore\'", "32 Address 4377505147 is in untracked image \'XCTestCore\'", "33 Address 4377465855 is in untracked image \'XCTestCore\'", "34 Address 4377464939 is in untracked image \'XCTestCore\'", "35 Address 4377471147 is in untracked image \'XCTestCore\'", "36 Address 4377242947 is in untracked image \'XCTestCore\'", "37 Address 4377469691 is in untracked image \'XCTestCore\'", "38 Address 4377360943 is in untracked image \'XCTestCore\'", "39 Address 4377468715 is in untracked image \'XCTestCore\'", "40 Address 4377168739 is in untracked image \'XCTestCore\'", "41 Address 4377373259 is in untracked image \'XCTestCore\'", "42 Address 4377372855 is in untracked image \'XCTestCore\'", "43 Address 4377371327 is in untracked image \'XCTestCore\'", "44 Address 4377360943 is in untracked image \'XCTestCore\'", "45 Address 4377371239 is in untracked image \'XCTestCore\'", "46 Address 4377371991 is in untracked image \'XCTestCore\'", "47 Address 4377168739 is in untracked image \'XCTestCore\'", "48 Address 4377373259 is in untracked image \'XCTestCore\'", "49 Address 4377372855 is in untracked image \'XCTestCore\'", "50 Address 4377371327 is in untracked image \'XCTestCore\'", "51 Address 4377360943 is in untracked image \'XCTestCore\'", "52 Address 4377371239 is in untracked image \'XCTestCore\'", "53 Address 4377371991 is in untracked image \'XCTestCore\'", "54 Address 4377168739 is in untracked image \'XCTestCore\'", "55 Address 4377373259 is in untracked image \'XCTestCore\'", "56 Address 4377372855 is in untracked image \'XCTestCore\'", "57 Address 4377371327 is in untracked image \'XCTestCore\'", "58 Address 4377360943 is in untracked image \'XCTestCore\'", "59 Address 4377371239 is in untracked image \'XCTestCore\'", "60 Address 4377371991 is in untracked image \'XCTestCore\'", "61 Address 4377168739 is in untracked image \'XCTestCore\'", "62 Address 4377175735 is in untracked image \'XCTestCore\'", "63 Address 4377360943 is in untracked image \'XCTestCore\'", "64 Address 4377175463 is in untracked image \'XCTestCore\'", "65 Address 4377597387 is in untracked image \'XCTestCore\'", "66 Address 4377597699 is in untracked image \'XCTestCore\'", "67 Address 4377349955 is in untracked image \'XCTestCore\'", "68 Address 4377212015 is in untracked image \'XCTestCore\'", "69 Address 4377349171 is in untracked image \'XCTestCore\'", "70 Address 4377170243 is in untracked image \'XCTestCore\'", "71 Address 4373075951 is in untracked image \'WooCommerceUITests-Runner\'", "72 Address 4373075923 is in untracked image \'WooCommerceUITests-Runner\'", "73 Address 6445997971 is in untracked image \'CoreFoundation\'", "74 Address 6445994599 is in untracked image \'CoreFoundation\'", "75 Address 6445972011 is in untracked image \'CoreFoundation\'", "76 Address 6445969943 is in untracked image \'CoreFoundation\'", "77 Address 6646265355 is in untracked image \'GraphicsServices\'", "78 Address 6523751063 is in untracked image \'UIKitCore\'", "79 Address 6523770419 is in untracked image \'UIKitCore\'", "80 Address 4373076483 is in untracked image \'WooCommerceUITests-Runner\'", "81 Address 4376501463 is in untracked image \'dyld_sim\'", "82 Address 4374696075 is in untracked image \'dyld\'", "83 Unable to determine owner (executable image) for address 16872595266455207935"])], history: Core.Trace.Span(section: "top", startAt: Optional(304929.4531096667), endAt: Optional(304962.7564847917), duration: Optional(33.30337512498954), detail: [:], children: []))])

[BuildkiteTestCollector] error: The data couldn’t be read because it is missing.
```

I can't make whether the issue is with the error format which doesn't report a test failure but a long list of "Address ... is in untracked image ..." or is due to some env var missing.

My next step will be to try the approach in the simplified environment of [TestPress](https://github.com/automattic/TestPress-Apple).